subprojects/build-events/src/main/java/org/gradle/build/event/BuildEventsListenerRegistry.java

```diff
@@ -37,7 +40,11 @@ public interface BuildEventsListenerRegistry {
      *
      * <p>The listener is automatically unsubscribed when the build finishes.</p>
      *
-     * @param listener The listener to receive events. This must be a {@link org.gradle.api.services.BuildService} instance, see {@link org.gradle.api.services.BuildServiceRegistry}.
+     * <p>In order to support Configuration Cache, the {@code listener} must be a provider returned <b>directly</b> from
+     * {@link BuildServiceRegistry#registerIfAbsent(String, Class, Action)} or {@link BuildServiceRegistration#getService()}
+     * </p>
+     *
+     * @param listener The listener to receive events. This must be a {@link org.gradle.api.services.BuildService} instance, see {@link BuildServiceRegistry}.
```

Comment:
💅 Nitpick: the `listener` parameter is not a build service instance by itself, but the provider must carry one.